include/vkwrap/vkwrap/sampler.h

```diff
@@ -0,0 +1,186 @@
+        ( std::optional<vk::Bool32>,             unnormalized_coordinates )
+    );
+    // clang-format on
+
+    using Setter = std::function<void( SamplerPartialInfo& )>;
+
+  private:
+    // clang-format off
+    Setter m_setter = []( auto& ){};
+    // clang-format on
+
+    SamplerPartialInfo m_partial;
+
+    SamplerPartialInfo makePartialInfo() const
+    {
+        SamplerPartialInfo partial{};
+
+        presetter( partial );
+        m_setter( partial );
+        partial.patchWith( m_partial );
+
+        return partial;
+    } // makePartialInfo
+
+    static float getMaxAnisotropy( vk::PhysicalDevice physical_device )
+    {
+        return physical_device.getProperties().limits.maxSamplerAnisotropy;
+    } // getMaxAnisotropy
+
+    static constexpr vk::SamplerCreateInfo k_sampler_initial_create_info{
+        // We don't use these specific fields.
+        .pNext = {},
+        .flags = {},
+
+        // We use images with only one mipmap.
+        .mipmapMode = vk::SamplerMipmapMode::eLinear,
+        .mipLodBias = 0.0f,
+
+        // compareOp would define behaviour of comparison.
+        .compareEnable = VK_TRUE,
+
+        // We use images with only one mipmap.
+        .minLod = 0.0f,
+        .maxLod = 0.0f,
+    };
+
+  public:
+    // clang-format off
+    static inline Setter presetter = []( auto& ){};
+    // clang-format on
+
+    SamplerBuilder() = default;
+
+    SamplerBuilder& withSetter( Setter setter ) &
+    {
+        m_setter = setter;
+        return *this;
+    } // withSetter
+
+    SamplerBuilder& withMagFilter( vk::Filter filter ) &
+    {
+        m_partial.mag_filter = filter;
+        return *this;
+    } // withMagFilter
+
+    SamplerBuilder& withMinFilter( vk::Filter filter ) &
+    {
+        m_partial.min_filter = filter;
+        return *this;
+    } // withMinFilter
+
+    SamplerBuilder& withAddressModeU( vk::SamplerAddressMode mode ) &
+    {
+        m_partial.address_mode_u = mode;
+        return *this;
+    } // withAddressModeU
+
+    SamplerBuilder& withAddressModeV( vk::SamplerAddressMode mode ) &
+    {
+        m_partial.address_mode_v = mode;
+        return *this;
+    } // withAddressModeV
+
+    SamplerBuilder& withAddressModeW( vk::SamplerAddressMode mode ) &
+    {
+        m_partial.address_mode_w = mode;
+        return *this;
+    } // withAddressModeW
+
+    SamplerBuilder& withAnisotropyEnable( vk::Bool32 is_enable ) &
+    {
+        m_partial.anisotropy_enable = is_enable;
+        return *this;
+    } // withAnisotropyEnable
+
+    SamplerBuilder& withBorderColor( vk::BorderColor color ) &
+    {
+        m_partial.border_color = color;
+        return *this;
+    } // withBorderColor
+
+    SamplerBuilder& withUnnormalizedCoordinates( vk::Bool32 is_unnormalized ) &
+    {
+        m_partial.unnormalized_coordinates = is_unnormalized;
+        return *this;
+    } // withUnnormalizedCoordinates
```

Comment:
Why Bool32 and not bool?

Reply:
Because SamplerCreateInfo::unnormalizedCoordinates is vk::Bool32.

Reply:
Still, it's a bool. The underlying Bool32 doesn't have to stick out from the function interface

Reply:
All builder's methods take vulkan objects. If I change Bool32 to bool, it would stick out the interface. So, I prefer all functions with vulkan-like arguments.